runtime/src/bank.rs

```diff
@@ -6648,6 +6649,111 @@ impl Bank {
             // AccountHash for modified accounts, and can stop the background account hasher.
             self.rc.accounts.accounts_db.stop_background_hasher();
         }
+
+        if new_feature_activations.contains(&feature_set::enshrine_slashing_program::id()) {
+            info!(
+                "Enshrine slashing program (SIMD-0204) activated at epoch {}",
+                self.epoch
+            );
+            self.enshrine_slashing_program();
+        }
+    }
+
+    fn get_build_hash_and_offsets(data: &[u8]) -> (usize, usize, Hash) {
+        let offset = bpf_loader_upgradeable::UpgradeableLoaderState::size_of_programdata_metadata();
+        let end_offset = data.iter().rposition(|&x| x != 0).map_or(0, |i| i + 1);
```

Comment:
Definitely a bit more efficient than the other impl :sweat_smile: 
nit: although this won't happen, just to be pedantically safe, can we fallback to `offset`?
```suggestion
        let end_offset = data.iter().rposition(|&x| x != 0).map_or(offset, |i| i + 1);
```